ostd/src/device/dispatcher/io_mem.rs

```diff
@@ -0,0 +1,234 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Io Memory access dispatcher.
+
+use alloc::{borrow::ToOwned, vec::Vec};
+use core::ops::Range;
+
+use log::{debug, info};
+use spin::Once;
+
+use crate::{
+    device::io_mem::IoMem,
+    mm::{CachePolicy, PageFlags},
+    util::vaddr_alloc::VirtAddrAllocator,
+};
+
+/// I/O memory dispatcher that allocates I/O access to devices via MMIO.
+pub struct IoMemDispatcher {
+    allocators: Once<Vec<VirtAddrAllocator>>,
+}
+
+impl IoMemDispatcher {
+    /// Get the IO memory access with `range`. If the range is not available, then the return value will be None.
```

Comment:
```suggestion
    /// Gets the I/O memory access for `range`.
    /// 
    /// If the range is not available, then the return value will be `None`.
```
As similar problems occur multiple times, please read [this](https://rust-lang.github.io/rfcs/1574-more-api-documentation-conventions.html#summary-sentence):
> In API documentation, the first line should be a single-line short sentence providing a summary of the code. This line is used as a summary description throughout Rustdoc’s output, so it’s a good idea to keep it short.
>
> The summary line should be written in third person singular present indicative form. Basically, this means write ‘Returns’ instead of ‘Return’.